### PR TITLE
[Feature] Chat scrolling, formatting and message types

### PR DESCRIPTION
**Pull request by palatter**
## Purpose

* Scroll fix for iOS 16
* Fetch messages
* Support additional message types
* Fixes for formatting of date header

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by JoshuaLai — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/MessageListViewModel.swift [hunk 1]
```diff
@@ -8,53 +8,70 @@ import Foundation
...
+                                        showTime: showTime,
+                                        isLocalUser: isLocalUser,
+                                        isConsecutive: isConsecutive)
+        case .participantsAdded, .participantsRemoved, .topicUpdated:
+            return SystemMessageViewModel(message: message,
+                                          showDateHeader: showDateHeader,
+                                          isConsecutive: false)
+        case .html:
+            return HtmlMessageViewModel(message: message, showDateHeader: showDateHeader, isConsecutive: isConsecutive)
+        case .custom(_): // Stub until finished
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/MessageListViewModel.swift:74]: do you happen to know what custom message we are expecting? Are those going to be video, gif, audio or something else?

**Review by jpeng-ms — APPROVED** [review]
Looks good, just have couple of questions not related to the ticket

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatConfiguration.swift [hunk 2]
```diff
@@ -14,7 +14,7 @@ struct ChatConfiguration {
     let credential: CommunicationTokenCredential
     let displayName: String?
     let diagnosticConfig: DiagnosticConfig
-    let pageSize: Int32 = 5
+    let pageSize: Int32 = 100
```
[the remark this answers is not in the document]
Comment (palatter) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/ChatCompositeOptions/ChatConfiguration.swift:17]: We have no plan for that at this time.

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/Message/HtmlMessageView.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+//
+
+import SwiftUI
+
+struct HtmlMessageView: View {
+    @StateObject var viewModel: HtmlMessageViewModel
+
+    var body: some View {
+        HStack {
+            Text(viewModel.message.content ?? "No Content")
```
[the remark this answers is not in the document]
Comment (palatter) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/Message/HtmlMessageView.swift:13]: This is just a stub for now.

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/MessageListViewModel.swift (hunk 1 above)
Comment (palatter) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/MessageListViewModel.swift:74]: I am not sure of the exact format for custom messages yet. This is just a stub for now.

**Pull request merged**